lab_3/main.py

```diff
@@ -20,6 +21,31 @@ def tokenize_by_sentence(text: str) -> tuple:
          (('_', 'h', 'e', '_'), ('_', 'i', 's', '_'), ('_', 'h', 'a', 'p', 'p', 'y', '_'))
          )
     """
+    if not isinstance(text, str):
+        return ()
+    patterns = ['ö', 'ü', 'ä', 'ß']
+    replacements = ['oe', 'ue', 'ae', 'ss']
+    sentences = re.split(r"[.!?] ?", text)
+    if '' in sentences:
+        sentences.remove('')
+    lst_of_sentences = [sentence.lower() for sentence in sentences]
+    new_sentence_tuple = []
+    for sentence in lst_of_sentences:
+        sentence = sentence.split(' ')
```

Comment:
Space is the default delimiter, no need to explicitly set it.

Reply:
done